test/test_retry.js

```diff
@@ -1,214 +1,181 @@
-'use strict';
+import test from 'ava';
+import util from 'util';
+import Promise from 'bluebird';
+import retryLib from '../index';
+
+const retryify = retryLib({
+  retries: 2,
+  timeout: 5, // ms
+  factor: 1.5,
+});
 
-var assert = require('assert');
-var util = require('util');
-var Promise = require('bluebird');
-var retryLib = require('../index');
+// MUST GET 100% COVERAGE (ノ._.)ノ
+retryLib();
 
-suite('Retryify', function() {
+test('no times, standard fn', async function(t) {
+  const addABC = retryify(function(a, b, c) {
+    return a + b + c;
+  }, {retries: 0});
 
-  var retryify;
+  const sum = await addABC(1, 2, 3);
+  t.is(sum, 6);
+});
 
-  suiteSetup(function() {
-    // low timeout for faster tests
-    retryify = retryLib({
-      retries: 2,
-      timeout: 5, // ms
-      factor: 1.5,
+test('no times, promise fn', async function(t) {
+  const addABC = retryify(function(a, b, c) {
+    return Promise.delay(5).then(function() {
+      return a + b + c;
     });
+  }, {retries: 0});
 
-    // MUST GET 100% COVERAGE (ノ._.)ノ
-    retryLib();
-  });
+  const sum = await addABC(1, 2, 3);
+  t.is(sum, 6);
+});
 
-  test('no times, standard fn', function() {
-    var addABC = retryify(function(a, b, c) {
-      return a + b + c;
-    }, { retries: 0 });
+test('once, error on first call, standard fn', async function(t) {
+  var retries = 1;
 
-    return addABC(1, 2, 3).then(function(sum) {
-      assert.equal(sum, 6);
-    });
-  });
-
-  test('no times, promise fn', function() {
-    var addABC = retryify(function(a, b, c) {
-      return Promise.delay(5).then(function() {
-        return a + b + c;
-      });
-    }, { retries: 0 });
+  const addFail = retryify(function(a, b, c) {
+    if (retries > 0) {
+      retries -= 1;
+      throw new Error('Oh no! The promise failed :0');
+    } else {
+      return a + b + c;
+    }
+  }, {retries});
 
-    return addABC(1, 2, 3).then(function(sum) {
-      assert.equal(sum, 6);
-    });
-  });
+  const sum = await addFail(1, 2, 3);
+  t.is(sum, 6);
+});
 
-  test('once, error on first call, standard fn', function() {
-    var retries = 1;
+test('once, error on first call, promise fn', async function(t) {
```

Comment:
if there's an `async` does there have to be an `await` somewhere? or is that what ava handles?